src/Android/Xamarin.Android/Samples/Symbology/SymbolsFromMobileStyle/SymbolsFromMobileStyle.cs

```diff
@@ -0,0 +1,677 @@
+
+        // The selected symbol layer info.
+        public SymbolLayerInfo SelectedSymbolInfo { get; set; }
+
+        // An event that notifies when the selected symbol has changed.
+        public event EventHandler<EventArgs> OnSymbolSelectionChanged;
+
+        // Constructor that takes the current activity and a list of symbol infos.
+        public SymbolListAdapter(Activity context, List<SymbolLayerInfo> symbolInfos) : base()
+        {
+            _ctx = context;
+            _symbolLayerInfos = symbolInfos;
+        }
+
+        // Get the symbol info at the specified position in the list.
+        public override SymbolLayerInfo this[int position]
+        {
+            get
+            {
+                return _symbolLayerInfos.ElementAt(position);
+            }
+        }
+
+        // Return the count of symbol infos in the list.
+        public override int Count
+        {
+            get
+            {
+                return _symbolLayerInfos.Count;
+            }
+        }
+
+        // Get an ID for the item at the specified position.
+        public override long GetItemId(int position)
+        {
+            return position;
+        }
+
+        // Create a view to display an item in the list (SymbolLayerInfo).
+        public override View GetView(int position, View convertView, ViewGroup parent)
+        {
+            // Create an ActivityListItem, since it contains an image to display the symbol swatch.
+            View listItem = _ctx.LayoutInflater.Inflate(Android.Resource.Layout.ActivityListItem, null);
+
+            // Display a symbol swatch if one is available, otherwise display the text "None".
+            if (_symbolLayerInfos.ElementAt(position).Image != null)
+            {
+                // Set the list item icon with the symbol swatch image.
+                listItem.FindViewById<ImageView>(Android.Resource.Id.Icon).SetImageBitmap(_symbolLayerInfos.ElementAt(position).Image);
+            }
+            else
+            {
+                // Display the text "None". Choosing this item will clear the symbol layer for that category.
+                listItem.FindViewById<TextView>(Android.Resource.Id.Text1).SetText("None", TextView.BufferType.Normal);
+            }
+
+            // Handle the list item click event.
+            listItem.Click += (sender, e) =>
+            {
+                // Get the clicked symbol layer info.
+                SelectedSymbolInfo = _symbolLayerInfos.ElementAt(position);
+
+                // Fire an event to notify clients that a symbol layer selection has changed.
+                OnSymbolSelectionChanged?.Invoke(this, new EventArgs());
+            };
+
+            return listItem;
+        }
+    }
+
+    // A class to contain information about a symbol from a style.
+    public class SymbolLayerInfo
+    {
+        // A bitmap preview of the symbol.
+        public Bitmap Image { get; private set; }
```

Comment:
I would check to see if these setters can be removed altogether

Reply:
It looks like changing to a field caused the binding to stop working on WPF; I recommend re-adding the getters, e.g. `public WinMedia.ImageSource ImageSrc { get; }`.

Reply:
Applies to UWP as well